### DISCUSSION TITLE
Flash memory reliability - MLC USB drive recommendations

### DISCUSSION THREAD

**Discussion by openoms** [raspiblitz/raspiblitz#3936] [Research]
Aiming to collect some of the more reliable flash drive options here.

Flash NAND chips (flash memory) seem to have 3 basic groups on the basis of how many bits they store per cell.

Single Level Cell (SLC) are basicly the common SSD-s which are more reliable and fast, but need more power and also more expensive than a USB flash drive.

There is a big difference between Multi Level Cell (quality pendrives) and Triple Level Cell drives (cheapest, but least reliable)

Rewrite cycles are respectively around:
80-100k for SLC
8-10k for MLC
3-5k for TLC (not suitable for constant rewrites even for months)

We should try to indentify some known MLC USB drives for the RAID setup otherwise there will be write errors encountered often.

Good summary here:
https://www.mydigitaldiscount.com/everything-you-need-to-know-about-slc-mlc-and-tlc-nand-flash.html
Software to test (only found for windows so far):
ChipGenius
USBFlashInfo

One known likely SLC drive I use:
Sandisk Extreme Pro USB  39 USD from China https://www.aliexpress.com/item/33048558218.html

**Comment by openoms** [comment]
SecretKey 32GB MLC USB 21.9 GBP https://www.amazon.co.uk/Drive%EF%BC%8CSecretKey-High-Speed-Storage-Pictures-Security/dp/B07Q1BP91Q

Mushkin impact 64GB MLC USB 23.29 USD https://www.amazon.co.uk/Mushkin-MKNUFDIM64GB-Impact-3-0-Black-Player/dp/B01A69S786

Transcend 32GB JF750 MLC USB30 BLACK 27 GBP https://www.ebay.co.uk/itm/32GB-JF750-MLC-USB30-BLACK/372877802051

**Comment by rootzoll** [comment]
This is good research to have once we make the Raid USB thumb drive part of the official shopping list - targeted for v1.5 release.

**Comment by rootzoll** [comment]
BTFS Raid will still be experimental in v1.5 ... moving issue to v1.6

**Comment by rootzoll** [comment]
BTFS Raid will still be experimental in v1.6 ... moving issue to Future Release

**Comment by openoms** [comment]
Took some measurements and found that even the USB 2.0 32 GB pendrive had power spikes up to 0.5 Amper when writing. 
The Sandisk Extreme Pro USB 128 GB SSD did not reach 0.5 Amper, but has a higher sustained power requirement ~ 0.25-0.3 Amper.
A 2 TB 2.5" Toshiba HDD was using 0.5-0.6  Amper during load (labelled to take need up to 1.0A).

For now the Sandisk Extreme Pro USB 128 GB SSD runs stable in a busy node with many services active.
Attached to this build: https://github.com/rootzoll/raspiblitz/issues/905#issue-536472177 and powered only through the USBC (3A PSU).

**Comment by xanoni** [comment]
FWIW, I have an old SanDisk Extreme USB 3.0 (32GiB) that I found among other old USB sticks. However, unlike the other USB sticks, this one is very very fast, similar to my SSDs (several hundred MiB/sec). I can't talk about reliability, but so far no issues.
